daml/Tests/Distribution/Auction.daml

```diff
@@ -0,0 +1,252 @@
+
+  -- Kick off the auction
+  (auctionCid, siCids) <- auctionAgent `submit` exerciseCmd auctionServiceCid Auction.ProcessAuction with ..
+
+  -- Assertions
+  length siCids === 4
+  auction <- fromSome <$> queryContractId auctionAgent auctionCid
+  auction.status === Auction.FullyAllocated with finalPrice = 10.0
+  bids <- query @Bid.Bid auctionAgent
+  length (filter ((\bid -> bid.status == Bid.FullAllocation with price = 10.0) . snd) bids) === 3
+  length (filter ((\bid -> bid.status == Bid.PartialAllocation with price = 10.0; quantity = 400.0) . snd) bids) === 1
+
+  -- Settle
+  depositCids <- forA siCids \siCid -> auctionAgent `submit` exerciseCmd siCid Settle.Settle
+
+  -- Settlement Assertions
+  auction <- snd . head <$> query @Auction.Auction auctionAgent
+  issuerPayments <- queryFilter @AssetDeposit issuer.customer (\deposit -> deposit.account == issuer.awayAccount && deposit.asset.id.label == auction.quotedAssetId.label) >>= \deposits -> pure . snd $ unzip deposits
+  assertAuctionMatchesPayments auction issuerPayments
+  F.forA_ bids \(_, bid) -> do
+    case bid.status of
+      Bid.FullAllocation price -> do
+        assertAssetDeposits auction issuer originalDeposits price bid.details.quantity bid
+      Bid.PartialAllocation price quantity -> do
+        assertAssetDeposits auction issuer originalDeposits price quantity bid
+      _ -> pure ()
+    pure ()
+
+auctionNoBidsAboveFloorPrice : Script ()
+auctionNoBidsAboveFloorPrice = do
+  (parties, originalDeposits, issuer, auctionAgent, custodian, auctionCid, auctionServiceCid, bidServices) <- setup
+
+  -- Create our bids
+  now <- getTime
+  bidDetails@[aliceBidDetails, bobBidDetails, charlieBidDetails, daveBidDetails] <- forA [(1.0, 500.0, now), (2.0, 300.0, now), (3.0, 100.0, now), (4.0, 100.0, now)] \(price, quantity, time) -> pure $ Bid.Details with price; quantity; time
+  bidCids@[aliceBidCid, bobBidCid, charlieBidCid, daveBidCId] <- forA (zip3 parties bidServices bidDetails) \(party, bidServiceCid, details@Bid.Details{..}) ->
+      party.customer `submit` exerciseCmd bidServiceCid Bid.SubmitBid with
+          tradingAccount = party.tradingAccount
+          allocationAccount = party.allocationAccount
+          depositCid = party.assetDepositCid
+          ..
+
+  -- Add our lead bid to the Auction
+  issuer.customer `submit` exerciseCmd auctionServiceCid Auction.AddLeadBidder with submitter = issuer.customer; leadBidCid = head bidCids
+
+  -- Kick off the auction
+  (auctionCid, siCids) <- auctionAgent `submit` exerciseCmd auctionServiceCid Auction.ProcessAuction with ..
+
+  -- Assertions
+  siCids === []
+  auction <- fromSome <$> queryContractId auctionAgent auctionCid
+  auction.status === Auction.NoValidBids
+  bids <- query @Bid.Bid auctionAgent
+  all ((\bid -> bid.status == Bid.Invalid) . snd) bids === True
+
+assertAssetDeposits : Auction.Auction -> CustomerContracts -> [AssetDeposit] -> Decimal -> Decimal -> Bid.Bid -> Script ()
```

Comment:
Are you checking to see if rejected bids get their deposit back? I remember that was a bug in the bond-auction repo.

Reply:
I'm adding this test in the next PR